File: frontend/src/main/frontend/src/app/wishes/item.ts
export class Item {
  itemId: number;
  name: string;
  description: string;
  ownerLogin: string;
  bookerId: string;
  priority: string;
  imageFilepath: string;
  link: string;
  dueDate: string;
  likes: number;
  tags: string[] = [];
  like: boolean;

  clone() : Item {
    let itemClone = new Item();

    itemClone.itemId = this.itemId;
    itemClone.name = this.name;
    itemClone.description = this.description;
    itemClone.ownerLogin = this.ownerLogin;
    itemClone.bookerId = this.bookerId;
    itemClone.priority = this.priority;
    itemClone.imageFilepath = this.imageFilepath;
    itemClone.link = this.link;
    itemClone.dueDate = this.dueDate;
    itemClone.likes = this.likes;
    itemClone.tags = this.tags;
    itemClone.like = this.like;
    return itemClone;
  }
}

File: first-service/src/main/java/com/kpi/project/first/service/entity/Item.java
package com.meetup.meetup.entity;

import lombok.Data;

import javax.validation.constraints.Max;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.util.List;

@Data
public class Item {

    private int itemId;

    @NotBlank
    @Size(min = 3, max = 50)
    private String name;

    @Size(max = 1023)
    private String description;

    private String imageFilepath;

    private String link;

    private String dueDate;

    private int likes;

    List<String> tags;

    @NotNull
    private int ownerId;

    private int bookerId;

    private ItemPriority priority;

    private boolean isLike;
}
